File: PathfinderPlugin/src/main/java/org/bukkit/pathfinding/CraftPathfinderManager.java
package org.bukkit.pathfinding;

import org.bukkit.entity.*;

/**
 * Created by ysl3000
 */
public class CraftPathfinderManager implements PathfinderManager {
    @Override
    public Insentient getPathfindeGoalEntity(Creature creature) {
        return null;
    }

    @Override
    public Insentient getPathfinderGoalEntity(Flying flying) {
        return new CraftInsentient(flying);
    }

    @Override
    public Insentient getPathfinderGoalEntity(Ambient ambient) {
        return new CraftInsentient(ambient);
    }

    @Override
    public Insentient getPathfinderGoalEntity(Slime slime) {
        return new CraftInsentient(slime);
    }

    @Override
    public Insentient getPathfinderGoalEntity(EnderDragon enderDragon) {
        return new CraftInsentient(enderDragon);
    }

    @Override
    public PathfinderPlayer getPathfinderPlayer(Player player) {
        return new CraftPathfinderPlayer(player);
    }
}

File: Pathfinder_1_13_1/src/main/java/com/github/ysl3000/bukkit/pathfinding/craftbukkit/v1_13_R2/pathfinding/CraftPathfinderManager.kt
package com.github.ysl3000.bukkit.pathfinding.craftbukkit.v1_13_R2.pathfinding

import org.bukkit.entity.Ambient
import org.bukkit.entity.Creature
import org.bukkit.entity.EnderDragon
import org.bukkit.entity.Flying
import org.bukkit.entity.Player
import org.bukkit.entity.Slime

import com.github.ysl3000.bukkit.pathfinding.craftbukkit.v1_13_R2.entity.CraftInsentient
import com.github.ysl3000.bukkit.pathfinding.entity.Insentient
import com.github.ysl3000.bukkit.pathfinding.pathfinding.PathfinderManager
import com.github.ysl3000.bukkit.pathfinding.pathfinding.PathfinderPlayer

/**
 * Created by ysl3000
 */
class CraftPathfinderManager : PathfinderManager {

    override fun getPathfindeGoalEntity(creature: Creature): Insentient {
        return CraftInsentient(creature)
    }

    override fun getPathfinderGoalEntity(flying: Flying): Insentient {
        return CraftInsentient(flying)
    }

    override fun getPathfinderGoalEntity(ambient: Ambient): Insentient {
        return CraftInsentient(ambient)
    }

    override fun getPathfinderGoalEntity(slime: Slime): Insentient {
        return CraftInsentient(slime)
    }

    override fun getPathfinderGoalEntity(enderDragon: EnderDragon): Insentient {
        return CraftInsentient(enderDragon)
    }

    override fun getPathfinderPlayer(player: Player): PathfinderPlayer {
        return CraftPathfinderPlayer(player)
    }
}
